src/main/java/de/tum/in/www1/artemis/service/LectureService.java

```diff
@@ -76,16 +76,23 @@ public void filterActiveAttachmentUnits(Lecture lectureWithAttachmentUnits) {
     }
 
     /**
-     * Filter active attachments for a set of lectures.
+     * Filter active attachments for a set of lectures. All lectures must be from the same course.
      *
+     * @param course                  course all the lectures are from
      * @param lecturesWithAttachments lectures that have attachments
      * @param user                    the user for which this call should filter
      * @return lectures with filtered attachments
      */
-    public Set<Lecture> filterActiveAttachments(Set<Lecture> lecturesWithAttachments, User user) {
+    public Set<Lecture> filterVisibleLecturesWithActiveAttachments(Course course, Set<Lecture> lecturesWithAttachments, User user) {
+        if (!authCheckService.isOnlyStudentInCourse(course, user)) {
```

Comment:
Can this be called with a user who is not part of the course?

Reply:
I check for `isAtLeastTeachingAssistantInCourse` now.